In [49]:
import csv

In [50]:
opened_file = open("guns.csv")

In [51]:
loaded_file = csv.reader(opened_file)

In [52]:
data = list(loaded_file)

In [53]:
data[0:5]

[['',
  'year',
  'month',
  'intent',
  'police',
  'sex',
  'age',
  'race',
  'hispanic',
  'place',
  'education'],
 ['1',
  '2012',
  '01',
  'Suicide',
  '0',
  'M',
  '34',
  'Asian/Pacific Islander',
  '100',
  'Home',
  '4'],
 ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'],
 ['3',
  '2012',
  '01',
  'Suicide',
  '0',
  'M',
  '60',
  'White',
  '100',
  'Other specified',
  '4'],
 ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4']]

In [54]:
headers = data[0]

In [55]:
data = data[1:]

In [56]:
headers

['',
 'year',
 'month',
 'intent',
 'police',
 'sex',
 'age',
 'race',
 'hispanic',
 'place',
 'education']

In [57]:
data[0:5]

[['1',
  '2012',
  '01',
  'Suicide',
  '0',
  'M',
  '34',
  'Asian/Pacific Islander',
  '100',
  'Home',
  '4'],
 ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'],
 ['3',
  '2012',
  '01',
  'Suicide',
  '0',
  'M',
  '60',
  'White',
  '100',
  'Other specified',
  '4'],
 ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4'],
 ['5',
  '2012',
  '02',
  'Suicide',
  '0',
  'M',
  '31',
  'White',
  '100',
  'Other specified',
  '2']]

In [58]:
years = []

In [59]:
for info in data:
    years.append(info[1])

In [60]:
years

['2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 '2012',
 

In [61]:
year_counts = {}

In [62]:
for year in years:
    if year not in year_counts:
        year_counts[year] = 1
    else:
        year_counts[year] += 1
year_counts

{'2012': 33563, '2013': 33636, '2014': 33599}

In [63]:
import datetime

In [64]:
dates = []

In [65]:
for info in data:
    int_year = int(info[1])
    int_month = int(info[2])
    date = datetime.datetime(year = int_year, month = int_month, day = 1)
    dates.append(date)

In [66]:
dates[0:5]

[datetime.datetime(2012, 1, 1, 0, 0),
 datetime.datetime(2012, 1, 1, 0, 0),
 datetime.datetime(2012, 1, 1, 0, 0),
 datetime.datetime(2012, 2, 1, 0, 0),
 datetime.datetime(2012, 2, 1, 0, 0)]

In [67]:
date_counts = {}

In [68]:
for date in dates:
    if date not in date_counts:
        date_counts[date] = 1
    else:
        date_counts[date] += 1

In [69]:
date_counts

{datetime.datetime(2012, 1, 1, 0, 0): 2758,
 datetime.datetime(2012, 2, 1, 0, 0): 2357,
 datetime.datetime(2012, 3, 1, 0, 0): 2743,
 datetime.datetime(2012, 4, 1, 0, 0): 2795,
 datetime.datetime(2012, 5, 1, 0, 0): 2999,
 datetime.datetime(2012, 6, 1, 0, 0): 2826,
 datetime.datetime(2012, 7, 1, 0, 0): 3026,
 datetime.datetime(2012, 8, 1, 0, 0): 2954,
 datetime.datetime(2012, 9, 1, 0, 0): 2852,
 datetime.datetime(2012, 10, 1, 0, 0): 2733,
 datetime.datetime(2012, 11, 1, 0, 0): 2729,
 datetime.datetime(2012, 12, 1, 0, 0): 2791,
 datetime.datetime(2013, 1, 1, 0, 0): 2864,
 datetime.datetime(2013, 2, 1, 0, 0): 2375,
 datetime.datetime(2013, 3, 1, 0, 0): 2862,
 datetime.datetime(2013, 4, 1, 0, 0): 2798,
 datetime.datetime(2013, 5, 1, 0, 0): 2806,
 datetime.datetime(2013, 6, 1, 0, 0): 2920,
 datetime.datetime(2013, 7, 1, 0, 0): 3079,
 datetime.datetime(2013, 8, 1, 0, 0): 2859,
 datetime.datetime(2013, 9, 1, 0, 0): 2742,
 datetime.datetime(2013, 10, 1, 0, 0): 2808,
 datetime.datetime(2013, 11,

In [70]:
sex_counts = {}

In [71]:
for info in data:
    sex = info[5]
    if sex not in sex_counts:
        sex_counts[sex] = 1
    else:
        sex_counts[sex] += 1

In [72]:
sex_counts

{'F': 14449, 'M': 86349}

In [73]:
race_counts = {}

In [74]:
for info in data:
    race = info[7]
    if race not in race_counts:
        race_counts[race] = 1
    else:
        race_counts[race] += 1

In [75]:
race_counts

{'Asian/Pacific Islander': 1326,
 'Black': 23296,
 'Hispanic': 9022,
 'Native American/Native Alaskan': 917,
 'White': 66237}

Well, when I checked the month to month difference in gun deaths, it seems that more deaths happen in summer months...June and July...this could be related to it being hotter, school being out and teens and such having too much time on their hands, or some other reasons that don't come immediatly to my mind right now. I also learned that WOW, there are way way more gun deaths by Men than Women. This tells me that Men have a violence and or mental health problem...as I suspected. To me, this has always meant that Women deal with their emotions in a healthier manner than Men. And as a society, it's not as OK for Men to be emotional...it is seen as weakness. And if you don't deal with your emotions, they get tied up and it becomes toxic. Lastly, I see that the most gun deaths, w/ regard to Race, come from Black and White people. Now, the number for White people is way higher, but I think that might be because there are so many more White people than Black people, so I would need to see those as percentages of their respective wholes to gain true insight.

In [76]:
opened_census_file = open("census.csv")

In [77]:
loaded_census_file = csv.reader(opened_census_file)

In [78]:
census = list(loaded_census_file)

In [79]:
census

[['Id',
  'Year',
  'Id',
  'Sex',
  'Id',
  'Hispanic Origin',
  'Id',
  'Id2',
  'Geography',
  'Total',
  'Race Alone - White',
  'Race Alone - Hispanic',
  'Race Alone - Black or African American',
  'Race Alone - American Indian and Alaska Native',
  'Race Alone - Asian',
  'Race Alone - Native Hawaiian and Other Pacific Islander',
  'Two or More Races'],
 ['cen42010',
  'April 1, 2010 Census',
  'totsex',
  'Both Sexes',
  'tothisp',
  'Total',
  '0100000US',
  '',
  'United States',
  '308745538',
  '197318956',
  '44618105',
  '40250635',
  '3739506',
  '15159516',
  '674625',
  '6984195']]

In [80]:
mapping = {
    "Asian/Pacific Islander": 15834141,
    "Black": 40250635,
    "Native American/Native Alaskan": 3739506,
    "Hispanic": 44618105,
    "White": 197318956
}

In [94]:
race_per_hundredk = {}

In [95]:
for k,v in race_counts.items():
    race_per_hundredk[k] = (v / mapping[k]) * 100000

In [96]:
race_per_hundredk

{'Asian/Pacific Islander': 8.374309664161762,
 'Black': 57.8773477735196,
 'Hispanic': 20.220491210910907,
 'Native American/Native Alaskan': 24.521955573811088,
 'White': 33.56849303419181}

In [97]:
intents = [ x[3] for x in data ]

In [98]:
intents

['Suicide',
 'Suicide',
 'Suicide',
 'Suicide',
 'Suicide',
 'Suicide',
 'Undetermined',
 'Suicide',
 'Accidental',
 'Suicide',
 'Suicide',
 'Suicide',
 'Suicide',
 'Suicide',
 'Homicide',
 'Suicide',
 'Suicide',
 'Suicide',
 'Homicide',
 'Suicide',
 'Suicide',
 'Suicide',
 'Homicide',
 'Suicide',
 'Homicide',
 'Homicide',
 'Suicide',
 'Suicide',
 'Suicide',
 'Homicide',
 'Suicide',
 'Suicide',
 'Suicide',
 'Suicide',
 'Homicide',
 'Undetermined',
 'Suicide',
 'Suicide',
 'Suicide',
 'Suicide',
 'Suicide',
 'Homicide',
 'Suicide',
 'Suicide',
 'Suicide',
 'Suicide',
 'Suicide',
 'Homicide',
 'Homicide',
 'Homicide',
 'Suicide',
 'Suicide',
 'Suicide',
 'Undetermined',
 'Suicide',
 'Suicide',
 'Accidental',
 'Homicide',
 'Suicide',
 'Homicide',
 'Suicide',
 'Homicide',
 'Suicide',
 'Suicide',
 'Suicide',
 'Suicide',
 'Suicide',
 'Suicide',
 'Suicide',
 'Suicide',
 'Suicide',
 'Accidental',
 'Suicide',
 'Accidental',
 'Suicide',
 'Undetermined',
 'Suicide',
 'Suicide',
 'Suicide',
 'Homi

In [99]:
races = [ x[7] for x in data ]

In [100]:
races

['Asian/Pacific Islander',
 'White',
 'White',
 'White',
 'White',
 'Native American/Native Alaskan',
 'White',
 'Native American/Native Alaskan',
 'White',
 'Black',
 'White',
 'Native American/Native Alaskan',
 'White',
 'Native American/Native Alaskan',
 'White',
 'Native American/Native Alaskan',
 'White',
 'Native American/Native Alaskan',
 'Asian/Pacific Islander',
 'White',
 'Native American/Native Alaskan',
 'White',
 'Black',
 'White',
 'White',
 'White',
 'Native American/Native Alaskan',
 'White',
 'White',
 'White',
 'Native American/Native Alaskan',
 'White',
 'Native American/Native Alaskan',
 'White',
 'White',
 'White',
 'Native American/Native Alaskan',
 'Native American/Native Alaskan',
 'White',
 'White',
 'Native American/Native Alaskan',
 'Asian/Pacific Islander',
 'White',
 'Native American/Native Alaskan',
 'Native American/Native Alaskan',
 'White',
 'White',
 'White',
 'Native American/Native Alaskan',
 'Asian/Pacific Islander',
 'White',
 'White',
 'Native Ame

In [101]:
homicide_race_counts = {}

In [102]:
for i, race in enumerate(races):
        if race not in homicide_race_counts:
            homicide_race_counts[race] = 0
        if intents[i] == "Homicide":
            homicide_race_counts[race] += 1     

In [103]:
homicide_race_counts

{'Asian/Pacific Islander': 559,
 'Black': 19510,
 'Hispanic': 5634,
 'Native American/Native Alaskan': 326,
 'White': 9147}

In [104]:
for k,v in homicide_race_counts.items():
    homicide_race_counts[k] = (v / mapping[k]) * 100000

In [105]:
homicide_race_counts

{'Asian/Pacific Islander': 3.530346230970155,
 'Black': 48.471284987180944,
 'Hispanic': 12.627161104219914,
 'Native American/Native Alaskan': 8.717729026240365,
 'White': 4.6356417981453335}

#findings: 
    The first thing that I noticed through analyzing and breaking down this data was the rise in gun deaths in the summer months. I presume this might have something to do with kids around the ages of 15-25 being out of school and having too much time on their hands. It also could be possible that these same kids are in gangs or into drugs. Also, I think there is a study somewhere about the effect of heat (summer months) which probably doesn't help when combined with violence and anger. 
    The second thing that I noticed was the absolutely staggering, but not surprising, difference in the amount of gun deaths by women (14449) and men (86349). This is madness. I posit that this is largely due to two things...two things that sort of go hand-in-hand. One is that I think women are more emotionally healthy than men. Women talk about their feelings and express their feelings, and men just shove everything down and away. Our male culture says that it's not OK to be weak and vulnerable and that you are a "pussy" or "fag" if you show emotion. Secondly, there is a lot of pressure on men to "succeed" and go out into the world and be successful and make money and provide (although times are-a-changing, this weight had been mostly carried by men). 
    The third even more staggering finding was the amount of homicide gun deaths per hundred-thousand person of each race category. It's terribly sad to see that whereas White and Asian/Pacific Islander have rates of around 3-5 gun deaths per hundred thousand people, Black people are sitting at a rate of 48-49 gun deaths per hundred-thousand people. This is not OK. We have serious work to do here. 
    The last thing that I thought was interesting and wanted to illuminate and discuss was the amount of total (homicide and suicide) gun deaths by race as it relates to Asian/ Pacific Islanders vs. all other race categories. Whereas White, Black, Hispanic are sitting at 20-58 gun deaths per hundred-thousand people, Asian/Pacific Islanders are sitting at a (positively) staggering 3 gun deaths per hundred-thousand. Clearly we all have something to learn from the culture, values, way of life, family dynamics, etc. of Asian/Pacific Islander people. We should look into this and learn from them!!!

#next steps: It would be interesting to look into this data further to see how education level correlates to gun deaths in total, but also homicide vs. suicide. I would assume that lower education would be higher amount of gun deaths, but what is interesting to think about is that I think that there might be higher suicide, potentially, for higher education. It just seems that in a way ignorance is bliss, and higher educated people are more in their heads, and when you are in your head all the time, it's easy to get depressed, anxious, etc...from my experience. It also would be interesting to look at how location is correlated to gun deaths. I will come back to this project in the future and add in this data at that time. Stay tuned.

Thanks for reading. see ya! -Mark